### PR TITLE
fix trace id value, introduce traceparent

### PR DESCRIPTION
**Pull request by VladislavMorozovIDT**
# Use Activity.Current.TraceId as traceId, Activity.Current.Id as traceparent
https://w3c.github.io/trace-context/#design-overview

- [x] You've read the [Contributor Guide](https://github.com/dotnet/aspnetcore/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/dotnet/aspnetcore/blob/main/CODE-OF-CONDUCT.md).
- [x] You've included unit or integration tests for your change, where applicable.
- [x] You've included inline docs for your change, where applicable.
- [x] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

Fix traceId, add traceparent field

## Description

Fixes #44679

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=44680) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=44680) before we can accept your contribution.<br/><br/>:x: VladislavMorozovIDT [sign now](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=44680)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/aspnetcore?pullRequest=44680) it.</sub>

**Comment by adityamandaleeka** [comment]
@samsp-msft Is this the right pattern for us to be using for this? If so, do we need to update other places too?

**Comment by adityamandaleeka** [comment]
@samsp-msft Ping...

File: src/Mvc/Mvc.Core/src/Infrastructure/DefaultProblemDetailsFactory.cs [hunk 1]
```diff
@@ -92,12 +92,18 @@ private void ApplyProblemDetailsDefaults(HttpContext httpContext, ProblemDetails
...
         }
 
-        var traceId = Activity.Current?.Id ?? httpContext?.TraceIdentifier;
+        var traceId = Activity.Current?.TraceId.ToString() ?? httpContext?.TraceIdentifier;
         if (traceId != null)
         {
             problemDetails.Extensions["traceId"] = traceId;
         }
 
+        var traceParent = Activity.Current?.Id ?? httpContext?.TraceIdentifier;
```
Comment (samsp-msft) [inline src/Mvc/Mvc.Core/src/Infrastructure/DefaultProblemDetailsFactory.cs:101]: This seems wrong, I think it should be Activity.Current.ParentID or Activity.Current.ParentSpanID depending on the desired format - they are the same thing, just different representations.

And there probably should be a spanID to represent the current activity.

**Comment by samsp-msft** [comment]
This is pretty confusing, and really depends on what MVC is trying to track here. To follow the W3C standard there are 3 fields to track:
- TraceID - a global ID  for the entire operation end to end, across processes
- ParentSpanID - the ID for the parent (if it exists)
- SpanID - the ID for the current task

As long as all 3 get logged, you can build up the hierarchy of what happened.

Activity.Id is a .NET way of representing this hierarchy and is a string format of $"00-{traceId}-{spanId}-{flagsChars}"

So the correct answer here really depends on what ProblemDetails represents. As these are all set as extension properties, arguments could be made for going with either the Id as is currently used, or splitting it out into the 3 fields. Either way, I don't think what this PR is doing is the right fix as its confusing the parent and current span id's.

More details are in https://learn.microsoft.com/en-us/dotnet/core/diagnostics/distributed-tracing-concepts#activity-ids
> Parent-Child relationships between Activities in the distributed trace tree are established using unique IDs. .NET's implementation of distributed tracing supports two ID schemes: the W3C standard [TraceContext](https://www.w3.org/TR/trace-context/), which is the default in .NET 5+, and an older .NET convention called 'Hierarchical' that's available for backwards compatibility. [Activity.DefaultIdFormat](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activity.defaultidformat#system-diagnostics-activity-defaultidformat) controls which ID scheme is used. In the W3C TraceContext standard, every trace is assigned a globally unique 16-byte trace-id ([Activity.TraceId](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activity.traceid#system-diagnostics-activity-traceid)), and every Activity within the trace is assigned a unique 8-byte span-id ([Activity.SpanId](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activity.spanid#system-diagnostics-activity-spanid)). Each Activity records the trace-id, its own span-id, and the span-id of its parent ([Activity.ParentSpanId](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activity.parentspanid#system-diagnostics-activity-parentspanid)). Because distributed traces can track work across process boundaries, parent and child Activities may not be in the same process. The combination of a trace-id and parent span-id can uniquely identify the parent Activity globally, regardless of what process it resides in.

**Comment by adityamandaleeka** [comment]
@VladislavMorozovIDT @samsp-msft Checking back in after the holidays... sounds like this needs more consideration. Should we move the discussion to the issue and close this PR for now?

**Comment by mkArtakMSFT** [comment]
Thanks for your PR, @VladislavMorozovIDT .
Based on @samsp-msft 's comment this PR isn't in the shape we would like to take.
Closing it for now. If you wish, please continue the conversation on the linked issue.

**Pull request closed without merging**